react/week3/todolist/src/App.js

```diff
@@ -0,0 +1,217 @@
+        </div>
+    )
+}
+
+function DeadlineInput(props) {
+    const [startDate, setStartDate] = useState(null);
+    const date = new Date();
+
+    return (
+        <DatePicker
+            placeholderText="Click to select a date"
+            closeOnScroll={true}
+            selected={startDate}
+            onChange={(date) => {
+                setStartDate(date)
+                props.changeMethod(date, 'date')
+            }
+            }
+            name="deadline"
+            value={props.value}
+            minDate={date}
+            openToDate={date}
+            className="date-input"
+            autoComplete="off"
+        />
+    )
+}
+
+function TodoForm(props) {
+    const [inputTodo, setInputTodo] = useState({
+        description: "",
+        deadline: ""
+    });
+    function setNewInput(inputValue, inputName) {
+        if (inputName === 'date') {
+            setInputTodo((prev) => {
+                return ({ ...prev, deadline: inputValue.getFullYear() + "-" + ((inputValue.getMonth() < 10) ? "0" : "") + (inputValue.getMonth() + 1) + "-" + ((inputValue.getDate() < 10) ? "0" : "") + inputValue.getDate() })
+            })
+        } else if (inputName === 'description') {
+            const { name, value } = inputValue;
+            setInputTodo((prev) => {
+                return ({ ...prev, [name]: value })
+            })
+        }
+    }
+
+    const { addMethod } = props
+    return (
+        <div className="add-todo-container">
+            <h3>Add a todo</h3>
+            <div className="inputs-container">
+                <DescriptionInput changeMethod={setNewInput} value={inputTodo.description} />
+                <DeadlineInput changeMethod={setNewInput} value={inputTodo.deadline} />
+                <div className="add-button-container">
+                    <button onClick={() => {
+                        addMethod(inputTodo)
+                        setInputTodo({
+                            description: "",
+                            deadline: null
+                        })
+                    }} className="add-button">Add todo</button>
+                </div>
+            </div>
+        </div>
+
+    )
+}
+
+function TodoListContainer() {
+    const [todo, setTodo] = useState([]);
+    const [id, setId] = useState(4);
+
+    useEffect(() => {
+        fetch("https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw")
+            .then(data => data.json())
+            .then(apiTodo => setTodo(prev => apiTodo))
+    }, [])
+    function incrementId() {
+        setId(prev => prev + 1);
+    }
+
+    function addTodo(newTodo) {
+        let { description, deadline } = newTodo
+        if (description.length === 0) {
+            description = undefined;
+        }
+        if (deadline === "") {
+            alert("Set a valid due date")
+            return
+        }
```

Comment:
```suggestion
    if (!description.length || !deadline.length) {
      alert("Set a valid due date or/and description");
      return;
    } 
```
I would suggest checking both things.

Reply:
I left the description without an alert to try the proptype so it can send an error to the console if the description is not passed.